src/stream/src/common/table/state_table.rs

```diff
@@ -138,13 +142,23 @@ pub struct StateTableInner<
 
     /// Watermark cache
     watermark_cache: StateTableWatermarkCache,
+
+    /// Data Types
+    /// We will need to use to build data chunks from state table rows.
+    data_types: Vec<DataType>,
+
+    /// Output indices
+    /// Used for:
+    /// 1. Computing output_value_indices to ser/de replicated rows.
+    /// 2. Computing output pk indices to used them for backfill state.
+    output_indices: Vec<usize>,
```

Comment:
How will we handle `output_indices` on the write path?

Reply:
For now no replication, subsequent PR will handle that.